jujugui/static/gui/src/app/jujulib/identity.js

```diff
@@ -19,28 +19,40 @@ var module = module;
     Initializer.
 
     @function identity
-    @param url {String} The URL of the identity instance, including
-      scheme and port, and excluding the API version.
+    @param userStorage {Object} An object containing access to the identity url.
     @param bakery {Object} A bakery object for communicating with the identity
       instance.
     @returns {Object} A client object for making identity API calls.
   */
-  function identity(url, bakery) {
+  function identity(userStorage, bakery) {
     // Store the API URL (including version) handling missing trailing slash.
-    this.url = url.replace(/\/?$/, '/') + identityAPIVersion;
+    this.userStorage = userStorage;
     this.bakery = bakery;
   };
 
   identity.prototype = {
 
+    getIdentityURL: function() {
+      const identityURL = this.userStorage.identityURL();
+      if (identityURL === null) {
+        return null;
+      }
+      return `${identityURL}/${identityAPIVersion}`;
+    },
+
     /**
       Fetch the user data from the identity service
       @param {String} userName The user name to fetch the data for.
       @param {Function} callback
         Called with the call signature (error <Object>, userData <Object>).
     */
     getUser: function(userName, callback) {
-      return this.bakery.get(`${this.url}/u/${userName}`, null, (err, resp) => {
+      const identityURL = this.getIdentityURL();
+      if (identityURL === null) {
+        callback('no identity url available', null);
```

Comment:
s/url/URL/ this is likely to be user facing.

Reply:
done